src/macros.jl

```diff
@@ -630,6 +630,39 @@ macro with(d, body)
     esc(with_helper(d, body))
 end
 
+# TODO: New docstring for @subset
+astable_rhs_docs = """
+In transformations, it is also allowed to use `AsTable(cols)` to work with
+multiple columns at once, where the columns are grouped together in a
+`NamedTuple`. When `AsTable(cols)` appears in a transformation, no
+other columns may be referenced in the block.
```

Comment:
how are then symbols outside of `AsTable` treated - they error or are treated as symbols?

Reply:
Also can `AsTable` appear in the specification multiple times?

Also is something like `target = AsTable(nms)` valid? And what should/can `target` be then?

Reply:
Good questions. 

The error would be the same as `transform(df, [AsTable([:a, :b]), :c] => ...)`

```
julia> @select df :y = begin 
           nothing
           AsTable([:x1, :x2])
       end
ERROR: ArgumentError: Table returned but a single output column was expected
Stacktrace:
```

but the following works

```
julia> @select df $AsTable = begin 
           nothing
           AsTable([:x1, :x2])
       end
10×2 DataFrame
 Row │ x1         x2       
     │ Float64    Float64  
─────┼─────────────────────
   1 │ 0.383283   0.394874
   2 │ 0.173797   0.285121
   3 │ 0.0262305  0.472488
   4 │ 0.0129008  0.990346
   5 │ 0.368682   0.247991
   6 │ 0.0210636  0.296448
   7 │ 0.473996   0.130082
   8 │ 0.680537   0.159793
   9 │ 0.0541043  0.572126
  10 │ 0.93897    0.050982
```

I have to fix the parsing to avoid that `nothing` in there. Will do

Reply:
I suppose I should also special-case `AsTable` on the LHS. But I can do that later.

Reply:
OK - thank you!